pkg/blobinfocache/boltdb/boltdb.go

```diff
@@ -324,8 +336,52 @@ func (bdc *cache) appendReplacementCandidates(candidates []prioritize.CandidateW
 // If !canSubstitute, the returned candidates will match the submitted digest exactly; if canSubstitute,
 // data from previous RecordDigestUncompressedPair calls is used to also look up variants of the blob which have the same
 // uncompressed digest.
+//
+// If scope is nil, final result also contains blobs with lowest priority without locations that could be possibly be reused.
 func (bdc *cache) CandidateLocations2(transport types.ImageTransport, scope types.BICTransportScope, primaryDigest digest.Digest, canSubstitute bool) []blobinfocache.BICReplacementCandidate2 {
-	return bdc.candidateLocations(transport, scope, primaryDigest, canSubstitute, true)
+	resAllBlobs := []prioritize.CandidateWithTime{}
+	var uncompressedDigestValue digest.Digest // = ""
+	v2Output := true
+	resultPrioritized := bdc.candidateLocations(transport, scope, primaryDigest, canSubstitute, v2Output)
+	// Reprocess all blobs and append them to resultPrioritized in lower priority
+	if err := bdc.view(func(tx *bolt.Tx) error {
```

Comment:
Why is this copy&pasted instead of happening inside `candidateLocations`? https://github.com/containers/image/pull/1645#discussion_r961900263 was marked as resolved.

Or is that still WIP?

Or is there some reason why doing it inside `candidateLocations` is unviable, or worse than copy&pasting?

Reply:
Moved now, PTAL again.